build-tasks.yaml

```
@@ -0,0 +1,26 @@
+version: "3"
+
+tasks:
+  target:
+    internal: true
+    vars:
+      TARGET: "{{default \"all\" .TARGET}}"
+    requires:
+      vars:
+        - name: "TARGET"
+          enum:
+            - "all"
+            - "spider_client"
+            - "spider_worker"
+            - "spider_task_executor"
+            - "unitTest"
+            - "worker_test"
+    deps: ["::config-cmake-project"]
+    cmds:
+      - "cmake --build {{.G_BUILD_SPIDER_DIR}} --target {{.TARGET}} --parallel"
```

```suggestion
      - "cmake --build {{.G_BUILD_SPIDER_DIR}} --target {{.TARGET}} --parallel {{.numCPU}}"
```

It's a good idea to restrict the parallelism to the number of cores or else low-resource machines can run out of memory.

---

`{{numCPU}}` instead of `{{numCPU}}` works. `numCPU` template function is added at version [3.40.0](https://taskfile.dev/changelog/#v3400---2024-11-05). I'll bump the version requirement for task in doc.